### PR TITLE
Implement Profile Listings Page

### LINKED ISSUE #21: FE: Profile Listings page
List all the profiles on the platform (kind of like a discovery page)

### PR DESCRIPTION
**Pull request by rj13371**
### What this PR does :
Made the skeleton for profile listings page, I have made fake profile objects and am rendering them until I can complete the profile listings integration ticket. 

### Screenshots / Videos :
demo screenshot
demo screenshot mobile

### Any information needed to test this feature :
You can view the route on http://localhost:3000/profile-listings 
But normally we would go to a dynamic route from the landing page that submits a search query of location, date in, date out , then redirects you to this page with the passed in props. 

The search bar is yet to be implemented.

### Any issues with the current functionality (optional):
Still having trouble with MUI's way of writing css so right now the Cards are not responsive on wide viewports and look awkward with lots of whitespace on the right. 

useStyles does not behave the way I thought it was originally intended for IE MUI overrides most classes you place unless you specifically target them or add them to the app theme which I obviously do not want to do for one component. I found out that the sx prop actually overrides MUI when you want it to so I am using it on a few components.

### CODE REVIEW

File: client/src/pages/Profiles/ProfileListings/ProfileListing.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import { Link as RouterLink } from 'react-router-dom';
+import { PinDrop } from '@mui/icons-material';
+
+interface Props {
+  location: string;
+  dropIn: Date;
+  dropOff: Date;
+}
+
+//we have an interface Profile in another branch so this is just temporary, change this to import a Profile interface when its on main branch.
```
Comment (sundayezeilo) [inline client/src/pages/Profiles/ProfileListings/ProfileListing.tsx]: Remove comment, please.

File: client/src/pages/Profiles/ProfileListings/ProfileListing.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  address: string;
+  pay: string;
+  _id: string;
+  photo: string;
+}
+
+type Profiles = Profile[];
+
+export default function ProfileListing({}: Props): ReactElement {
+  //currently fake profiles for state
```
Comment (sundayezeilo) [inline client/src/pages/Profiles/ProfileListings/ProfileListing.tsx]: Comment not needed please

File: client/src/pages/Profiles/ProfileListings/ProfileListing.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  const [profiles, setProfiles] = useState<Profiles>([]);
+  const classes = useStyles();
+
+  useEffect(() => {
+    // use this later when we can integrate, this is an example of getting an API resp and then setting state with it
+    //   getProfiles().then((res) => {
+    //   });
```
Comment (sundayezeilo) [inline client/src/pages/Profiles/ProfileListings/ProfileListing.tsx]: You can make a note of this in  your todos but not suitable here as it makes our code untidy :smiley:

**Review by sundayezeilo — COMMENTED** [review]
Looks good! Please take a look at the few comments I left you and don't forget to link your ticket in your next PRs.
Happy coding!

File: client/src/pages/Profiles/ProfileListings/ProfileListing.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import { Link as RouterLink } from 'react-router-dom';
+import { PinDrop } from '@mui/icons-material';
+
+interface Props {
+  location: string;
+  dropIn: Date;
+  dropOff: Date;
+}
+
+interface Profile {
```
Comment (rajivtitus) [inline client/src/pages/Profiles/ProfileListings/ProfileListing.tsx]: Why don't we move this interface to the 'interface' directory

File: client/src/pages/Profiles/ProfileListings/ProfileListing.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      name: 'Roland Matheson',
+      title: 'Pet Lover',
+      description:
+        'I love dogcats Lorem ipsum dolor sit amet consectetur adipisicing elit. Perferendis assumenda omnis obcaecati, cumque, nisi animi quam voluptatum tempora enim illo at id iusto! Iure eveniet cum in a, accusamus earum!',
+      address: 'Tokyo, Japan',
+      pay: '$99/Hour',
+      _id: '61df3960442e349d9200000',
+      photo: 'fakeimage',
+    };
+    const fakeProfiles = [profile, profile, profile, profile, profile];
```
Comment (rajivtitus) [inline client/src/pages/Profiles/ProfileListings/ProfileListing.tsx:50]: 👍🏽

File: client/src/pages/Profiles/ProfileListings/ProfileListing.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      <Box margin={{ md: 10 }}>
+        <Typography className={classes.header} variant="h3" component="div">
+          {`Your search results`}
+        </Typography>
+
+        <Grid container spacing={2}>
+          {profiles.length > 0 &&
+            profiles.map((profile) => (
+              <Grid key={profile.userId} item md={'auto'}>
+                <Card className={classes.profileCard}>
```
Comment (rajivtitus) [inline client/src/pages/Profiles/ProfileListings/ProfileListing.tsx]: Good use of the MUI Card API 💯

File: client/src/pages/Profiles/ProfileListings/ProfileListing.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+                    <CardContent>
+                      <Typography gutterBottom variant="h4" component="div">
+                        {`${profile.name}`}
+                      </Typography>
+                      <Typography gutterBottom variant="h5" color="text.secondary" component="div">
+                        {`${profile.title}`}
+                      </Typography>
+
+                      <Rating name="half-rating" defaultValue={2.5} precision={0.5} />
```
Comment (rajivtitus) [inline client/src/pages/Profiles/ProfileListings/ProfileListing.tsx]:
```suggestion
                      <Rating name="half-rating" defaultValue={2.5} precision={0.5} readOnly />
```

**Review by rajivtitus — COMMENTED** [review]
Very good work with this Roland!
Let's also complete the following:
- Add text fields/date pickers for the search bar 
- Refine the padding for the UI
- Maybe use the substring method to only display a limited number of chars on each card
Thanks 👍🏽

**Comment by rj13371** [comment]
I refined the UI to look better on wider monitors, added a substring of 80 chars max, and added a placeholder form for location and date, but as I understand this is being made in the Dashboard integration ticket right? So this will be changed later?

**Comment by rajivtitus** [comment]
> I refined the UI to look better on wider monitors, added a substring of 80 chars max, and added a placeholder form for location and date, but as I understand this is being made in the Dashboard integration ticket right? So this will be changed later?

We just needed the search bar UI in this PR, it will be made functional in the `FS: Dashboard Ticket`

**Review by rajivtitus — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**